### PR TITLE
feat(scheduling): add staff schedule view API

### PR DESCRIPTION
**Pull request by nigelnindodev**
- Add GET /staff/schedule endpoint for authenticated staff to view their assignments within a date range
- Add StaffScheduleService with getMySchedule method
- Add findByStaffMemberAndDateRangeWithDetails to AssignmentRepository with full joins to shift, skill, and location
- Add location and skill relations to Shift and ShiftSkill entities
- Add StaffScheduleEntryDto and StaffScheduleQueryDto
- Add 9 integration tests covering date ranges, state filtering, chronological ordering, and boundary conditions

## Summary by CodeRabbit

* **New Features**
  * Staff schedule endpoints to retrieve detailed shift assignments for a date range, including assignment/shift IDs, state, start/end times, location name/timezone, and skill info; staff users can only view their own schedules.

* **Bug Fixes / Validation**
  * Date range validation: requests with startDate >= endDate are rejected.

* **Tests**
  * Integration tests for retrieval, date filtering (inclusive bounds), state exclusions, ordering, and access restrictions.

### CODE REVIEW

File: server/src/scheduling/services/staff-schedule.service.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { ClockService } from '../../common/clock/clock.service';
+
+@Injectable()
+export class StaffScheduleService {
+  private readonly logger = new Logger(StaffScheduleService.name);
+
+  constructor(
+    private readonly assignmentRepo: AssignmentRepository,
+    private readonly clockService: ClockService,
+  ) {}
```
[the remark this answers is not in the document]
Comment (nigelnindodev) [reply server/src/scheduling/services/staff-schedule.service.ts]: Minor. Can stay for now. No need to create issue.
